Gems/Atom/Tools/MaterialEditor/Code/CMakeLists.txt

```diff
@@ -82,3 +82,54 @@ else()
     message(FATAL_ERROR "Cannot set LY_CMAKE_TARGET define to MaterialEditor as the target doesn't exist anymore."
         " Perhaps it has been renamed")
 endif()
+
+# ---------------------------------------------
+# Atom Material Editor as module for O3DE
+# ---------------------------------------------
+
+ly_add_target(
+    NAME AtomMaterialEditor.Static STATIC
```

Comment:
`AtomMaterialEditor.Static` and `AtomMaterialEditor` targets only include a module and system component to listen to `O3DEMaterialEditor` notification and register the same window that the stand alone application uses.

`AtomMaterialEditor.Static` reuses all the same classes as standalone version.

Another gem can be created for these 2 targets to have them completely separate from MaterialEditor EXECUTABLE target. For now I just put it here for convenience of reusing the same classes.

I was thinking that it'd be nice if all the common classes (under Window, Document and Viewport folders) could be put into a library, so both AtomMaterialEditor and standalone MaterialEditor could use it.

Reply:
that sounds like a great idea